### DISCUSSION TITLE
Convex Border Radius

### DISCUSSION THREAD

**Discussion by jamessoncardozo** [tailwindlabs/tailwindcss#12271] [Help]
Hi Guys,

Is it possible to create an inverted border on an element using a negative border radius, similar to the example "rounded-l-[-9999px]"?

**Comment by wongjn** [comment]
It would not be possible with the `border-radius` property, as per [the CSS spec](https://www.w3.org/TR/css-backgrounds-3/#border-radius):

> Negative values are invalid.

**Reply by jamessoncardozo** [reply]
Yes, that's true. I had already seen that. However, I'm exploring if there's another way to create a convex border for my project's needs. The solutions I found involve a workaround using images or pseudo-elements. Wouldn't there be a more elegant way to achieve this?

**Reply by wongjn** [reply]
What criteria would a solution need to meet to be regarded as "elegant" for you?

**Reply by jamessoncardozo** [reply]
For me, perhaps using the simplicity of Tailwind through a class like `-rounded-full`, for instance, to reverse the border direction might be simpler than using an image as a workaround for achieving the negative border-radius effect.

It's just a question, really, as I'm not sure if it's something that Tailwind has already considered or will consider at some point. I couldn't find it in the Tailwind repository here.

**Reply by wongjn** [reply]
As you have eluded to, making concave rounded corners would take some gymnastics to work – there is no single declaration that would give the effect. Thus, the implementation would be fairly specific to the HTML page, so I'm not so sure that it would be suitable for a general utility library – you'd want to look at implementing in your own project specific to its context.

**Comment by 8mist** [comment]
Hi, a few days ago I released a tailwind plugin to do what I think you need.

[https://github.com/butterfail/tailwindcss-inverted-radius](https://github.com/butterfail/tailwindcss-inverted-radius)

**Reply by Stosan** [reply]
can you show an example usage for the concave. the repo doesn't do so

**Reply by Dennis4720** [reply]
^^^ please. @8mist

**Reply by wongjn** [reply]
There's the [usage section](https://github.com/8mist/tailwindcss-inverted-radius?tab=readme-ov-file#usage) which shows some examples. [Here's a quick Tailwind Play with some further examples](https://play.tailwindcss.com/6VVa4MzYwJ).
